Source/ACE/Factories/SpellObjectFactory.cs

```diff
@@ -0,0 +1,73 @@
+﻿using ACE.Entity;
+using ACE.Network.Enum;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ACE.Factories
+{
+
+    /// <summary>
+    /// Spell Factory for creating objects related to Spells.
+    /// </summary>
+    public class SpellObjectFactory
+    {
+        public static WorldObject CreateSpell(uint templateId, Position position, AceVector3 velocity, float friction, float electicity)
+        {
+
+            Entity.SpellEntity wo = new Entity.SpellEntity(position, velocity, friction, electicity);
+
+            // todo: impletment more advanced templating.
+            switch (templateId)
+            {
+                case 0:
+                    wo.SpellID = Spell.FlameBolt;
```

Comment:
yeah, as I said above, these values should be pushed into the MutableWorldObject constructor instead of being replaced after instantiation.